props/saber_sa22c_buttons.h

```diff
@@ -103,236 +165,250 @@ class SaberSA22CButtons : public PropBase {
-      case EVENTID(BUTTON_POWER, EVENT_CLICK_SHORT, MODE_ON):
-        SaberBase::DoBlast();
-        return true;
+// 1 button
+  case EVENTID(BUTTON_POWER, EVENT_FIRST_HELD_LONG, MODE_ON):
 #endif
-      case EVENTID(BUTTON_POWER, EVENT_DOUBLE_CLICK, MODE_ON):
-        if (millis() - activated_ < 500) {
-          if (SetMute(true)) {
-            unmute_on_deactivation_ = true;
-          }
-        }
-#if NUM_BUTTONS > 1
-        else {
-          SaberBase::DoForce();
-        }
+#ifdef SHTOK_GESTURE_IGNITION
+  case EVENTID(BUTTON_NONE, EVENT_TWIST, MODE_ON):
 #endif
-        return true;
-
-#if NUM_BUTTONS == 1
-        case EVENTID(BUTTON_POWER, EVENT_HELD_LONG, MODE_ON):
-        if (!SaberBase::Lockup()) {
+    if (!SaberBase::Lockup()) {
 #ifndef DISABLE_COLOR_CHANGE
-          if (SaberBase::GetColorChangeMode() != SaberBase::COLOR_CHANGE_MODE_NONE) {
-            // Just exit color change mode.
-            // Don't turn saber off.
-            ToggleColorChangeMode();
-            return true;
-          }
-#endif
-          Off();
-        }
+      if (SaberBase::GetColorChangeMode() != SaberBase::COLOR_CHANGE_MODE_NONE) {
+        // Just exit color change mode.
+        // Don't turn saber off.
+        ToggleColorChangeMode();
         return true;
+      }
 #endif
+      Off();
+    }
+    swing_blast_ = false;
+    return true;
 
+// 1 button Force and Color Change mode
+#if NUM_BUTTONS == 1
+  case EVENTID(BUTTON_NONE, EVENT_TWIST, MODE_ON | BUTTON_POWER):
 #ifndef DISABLE_COLOR_CHANGE
-#ifdef SA22C_NO_LOCKUP_HOLD
-    case EVENTID(BUTTON_POWER, EVENT_CLICK_SHORT, MODE_ON | BUTTON_AUX):
-      ToggleColorChangeMode();
-      break;
-#else //SA22C_NO_LOCKUP_HOLD
-    case EVENTID(BUTTON_AUX, EVENT_CLICK_SHORT, MODE_ON | BUTTON_POWER):
+    if (accel_.x < -0.15) {
       ToggleColorChangeMode();
-      break;
-#endif //SA22C_NO_LOCKUP_HOLD
-#endif //DISABLE_COLOR_CHANGE
-#if NUM_BUTTONS > 1
-      case EVENTID(BUTTON_POWER, EVENT_HELD_MEDIUM, MODE_ON):
-      case EVENTID(BUTTON_POWER, EVENT_LATCH_OFF, MODE_ON):
-      case EVENTID(BUTTON_AUX, EVENT_LATCH_OFF, MODE_ON):
-      case EVENTID(BUTTON_AUX2, EVENT_LATCH_OFF, MODE_ON):
-#endif
-#if NUM_BUTTONS == 0
-      case EVENTID(BUTTON_NONE, EVENT_TWIST, MODE_ON):
+    } else {
+      SaberBase::DoForce();
+    }
+#else
+    SaberBase::DoForce();
 #endif
-#if NUM_BUTTONS != 1
-        if (!SaberBase::Lockup()) {
+    return true;
+#else
+  // 2 and 3 button Force effect
+  case EVENTID(BUTTON_POWER, EVENT_SECOND_CLICK_SHORT, MODE_ON):
+    SaberBase::DoForce();
+    return true;
+  // 2 and 3 button color change
 #ifndef DISABLE_COLOR_CHANGE
-          if (SaberBase::GetColorChangeMode() != SaberBase::COLOR_CHANGE_MODE_NONE) {
-            // Just exit color change mode.
-            // Don't turn saber off.
-            ToggleColorChangeMode();
-            return true;
-          }
+  case EVENTID(BUTTON_AUX, EVENT_FIRST_CLICK_SHORT, MODE_ON | BUTTON_POWER):
```

Comment:
When performing hold power and toggle aux for color change mode, I'm getting a blast effect when the aux button is released.  Any suggestions on how I might stop that from happening, without compromising my ability to have a double-click happen WITHOUT the first click also producing a blast?

Reply:
I updated button_base to not emit saved events for events that have already triggered an action.
I think that should fix the problem, please let me know if it works or not.

Reply:
That did it, thanks!